test/system/252-quadlet.bats

```diff
@@ -1599,4 +1599,66 @@ EOF
     run_podman rmi $untagged_image:latest $built_image $(pause_image)
     run_podman network rm podman-default-kube-network
 }
+
+@test "quadlet - dropin files" {
+    local tmp_dir=${PODMAN_TMPDIR}/dropin-$(random_string)
+
+    local quadlet_file=${tmp_dir}/truncated-$(random_string).container
+
+    local toplevel_dropin_dir=${tmp_dir}/container.d
+    local truncated_dropin_dir=${tmp_dir}/truncated-.container.d
+    local quadlet_file_dropin_dir=${quadlet_file}.d
+
+    local base_dropin_file=10-base.conf
+    local custom_dropin_file=50-custom.conf
+    local override_dropin_file=99-override.conf
+
+    mkdir -p ${toplevel_dropin_dir} ${truncated_dropin_dir} ${quadlet_file_dropin_dir}
+
+    # This should be applied
+    cat > ${toplevel_dropin_dir}/${base_dropin_file} <<EOF
+[Unit]
+Description=Test File for Dropin Configuration
+EOF
+    # This should NOT be applied
+    cat > ${toplevel_dropin_dir}/${override_dropin_file} <<EOF
+[Install]
+WantedBy=default.target
+EOF
+    # This should be applied
+    cat > ${truncated_dropin_dir}/${custom_dropin_file} <<EOF
+[Container]
+ContainerName=truncated-dropins
+EOF
+    # This should NOT be applied
+    cat > ${truncated_dropin_dir}/${override_dropin_file} <<EOF
+[Service]
+Restart=always
+
+[Install]
+WantedBy=multiuser.target
+EOF
+    # This should be applied
+    cat > ${quadlet_file_dropin_dir}/${override_dropin_file} <<EOF
+[Service]
+RestartSec=60s
+EOF
+
+    cat > $quadlet_file <<EOF
+[Container]
+Image=$IMAGE
+EOF
+
+    QUADLET_UNIT_DIRS="${tmp_dir}" run \
+                    timeout --foreground -v --kill=10 $PODMAN_TIMEOUT \
+                    $QUADLET --dryrun
```

Comment:
Regarding your suggestion in `run_quadlet`:
```
run cat $UNIT_DIR/$QUADLET_SERVICE_NAME
assert $status -eq 0 "Could not cat $UNIT_DIR/$QUADLET_SERVICE_NAME"
echo "$output"
```
Would setting the output of `run cat` to an env var be appropriate? I noticed other global vars are set in the function, and this could allow a more direct way of accessing the content. Something like `QUADLET_SERVICE_CONTENT`. My main reasoning is it would prevent future changes to `run_quadlet`, where `$output` is changed, from impacting the test